src/contexts/LikedContext.tsx

```diff
@@ -0,0 +1,74 @@
+import {
+  createContext,
+  useCallback,
+  useState,
+  useContext,
+  useEffect,
+  useRef,
+} from "react";
+import { getData, storeData } from "../utils";
+
+interface LikeContextType {
+  likeList: number[];
+  toggleLike: (id: number) => void;
+  isLiked: (id?: number) => boolean;
+}
+
+const LikeContext = createContext<LikeContextType>({
+  likeList: [],
+  toggleLike: () => {},
+  isLiked: () => false,
+});
+
+const LikeContextProvider = ({ children }: { children: React.ReactNode }) => {
+  const isFirstRender = useRef(true);
+  const [likeList, setLikedList] = useState<number[]>([]);
+  useEffect(() => {
+    (async () => {
+      const initialLikes = await getData<number[]>("likeList");
```

Comment:
Storing `likeList` as some const would be less error prone (so you will have defined this string in one place)